components/calendar/src/calendar_arithmetic.rs

```diff
@@ -16,14 +18,21 @@ use core::fmt::Debug;
 use core::hash::{Hash, Hasher};
 use core::ops::RangeInclusive;
 
-/// The range ±2²⁷. We use i32::MIN since it is -2³¹
-///
-/// This range is currently global, and applied to both era years and
-/// extended years, but may be replaced with a per-calendar check in the future.
+/// This is checked by constructors. Internally we don't care about this invariant.
+pub const VALID_YEAR_RANGE: RangeInclusive<i32> = -1_000_000..=1_000_000;
+
+/// This is a fundamental invariant of `ArithmeticDate` and by extension all our
+/// date types. Because this range slightly exceeds the [`VALID_YEAR_RANGE`], only
+/// the valid year range is checked in constructors. Only the `Date::from_rata_die`
+/// constructor actually uses this, but for clamping instead of erroring.
 ///
-/// <https://github.com/unicode-org/icu4x/issues/7076>
-const VALID_YEAR_RANGE: RangeInclusive<i32> = (i32::MIN / 16)..=-(i32::MIN / 16);
+/// The longest average year length is 365.25 (Julian, Coptic). We round up to 366 to make
+/// it round and capture year offsets from the Gregorian calendar, which underlies RD.
+pub const VALID_RD_RANGE: RangeInclusive<RataDie> =
```

Comment:
thought: we can be sure that all valid extended years are within VALID_YEAR_RANGE because the largest offset from 1970 is the Hebrew calendar (-5000) which is nothing at this scale.

suggestion (nb): this comment explains how this works for year length but not for far-from-1970 epochs.

Reply:
Ah. Same point stands, since we test VALID_YEAR_RANGE on extended years, not CE years.

Reply:
Ah, I see, I didn't realize that that was what that talked about.